### PR TITLE
chore(card view demos): update to new selectable card variant

### LINKED ISSUE #9280: Update Card View and Primary-Detail demos to utilize new v5 card designs
**Describe the enhancement or change**

I'm noticing that the Card View and Primary-Detail demos that use selectable/clickable cards are still using the PF4 legacy card designs. We should update when possible.

This demo: https://www.patternfly.org/v4/demos/card-view and this one: https://www.patternfly.org/v4/demos/primary-detail#primary-detail-card-view

### PR DESCRIPTION
**Pull request by jenny-s51**
**What**: Closes #9280

### CODE REVIEW

**Review by kmcfaul — COMMENTED** [review]
Looks like the docs build is complaining about a label a11y violation for these, otherwise the changes lgtm

**Review by jenny-s51 — COMMENTED** [review]
Thank you @kmcfaul - updated

File: packages/react-core/src/demos/Card/Card.md
Comment (thatblindgeye) [inline packages/react-core/src/demos/Card/Card.md:662]: We should remove both the onKeyDown and onClick here and instead pass an onChange property to the `selectableActions` prop in CardHeader. onKeydown won't be needed with the onChange, and based on how we're suggesting a clickable and selectable card should behave we don't want to "select" the card when we click anywhere within it.

File: packages/react-core/src/demos/PrimaryDetail.md
Comment (thatblindgeye) [inline packages/react-core/src/demos/PrimaryDetail.md:536]: Similar to the above comment re: onKeyDown and onClick

File: packages/react-core/src/demos/PrimaryDetail.md
Comment (thatblindgeye) [inline packages/react-core/src/demos/PrimaryDetail.md:572]: Rather than calling onCardClick, could we add an onChange method to call instead? Might be the same functionality, but just to have a more appropriate name.

Though a question for @andrew-ronaldson . Do we want to trigger the drawer opening upon "selecting" a Card in this demo? Or should we add another clickable action somewhere that would do that (making the card title a button, adding an "open drawer" button somewhere, etc)?

**Review by thatblindgeye — CHANGES_REQUESTED** [review]

File: packages/react-core/src/demos/Card/Card.md
Comment (jenny-s51) [reply packages/react-core/src/demos/Card/Card.md:662]: Thanks Eric, this is a great idea.  I thought we would want the full card to be selectable as it already had been, but due to the presence of the kebab it makes sense for only the checkbox area to be selectable now.

I added these updates in the [Card View refactor PR](https://github.com/patternfly/patternfly-react/pull/9359/commits/758641fea1d3e486ed9ffd3e077af2a4fb9ef166) because passing `onChange` this way lends itself better to TS. and was causing some issues when used within a class component

File: packages/react-core/src/demos/PrimaryDetail.md
Comment (jenny-s51) [reply packages/react-core/src/demos/PrimaryDetail.md:572]: Good points - will be working on a Primary Detail refactor (#9363) and plan to add this change to the TS conversion PR based on the TS component syntax.

I have some thoughts for your question as well... I think we would want to toggle the drawer when clicking a card, but it would be mutually exclusive from "selecting". I.e. when the checkbox is clicked, the drawer is not toggled, but if a user clicks anywhere else on the card, the drawer is toggled.  That also aligns with the updated Card View behavior based on your feedback above. WDYT @thatblindgeye ?

Comment (thatblindgeye) [reply packages/react-core/src/demos/PrimaryDetail.md:572]: There would probably be some issues with making the whole card clickable to trigger opening a drawer. The main thing I'm thinking of is how we would allow clicking the card via keyboard, as we wouldn't be able to wrap the card in a button since there's already interactive content inside of it. We could probably add several things to the card wrapper to do it (tabindex, an onClick, some role, and some labeling that signifies the card can be clicked to open a drawer + possibly the drawer's current expanded state), but that's adding a lot of ARIA and other attributes when we may not need to.

Another might be if there's more tweaking that has to be done if the card includes links (having to avoid opening the drawer on selection, kebab toggle click, and link click), though I imagine any steps taken to avoid triggering the drawer action on selection/kebab click would easily work for something like a link/some other action button.

Considering all of that I feel like it'd be best to utilize a separate button somewhere in the card. Definitely something design should weigh in on, though.

**Comment by mmenestr** [comment]
From a visual/interaction standpoint, this looks good to me - the only comment I have is that I'm not sure whether the intended design on the clickable+selectable card was for the card outline to darken when the checkbox is selected. I thought just the checkbox was meant to change states (?) That's what we have in our design guidelines anyway.

Referring to this:
image

I could be wrong though - going to double check with Tina who worked on the designs !!

**Comment by mmenestr** [comment]
Just double checked with her and she said the outline should be the same whether the card is checked or not!

**Review by jenny-s51 — COMMENTED** [review]
Thank you for investigating @mmenestr !

This implementation of Card View matches the Card View design guidelines https://staging.patternfly.org/patterns/card-view/design-guidelines where the behavior and styling is a bit different from the typical standalone cards.

Here is an example of the styling as shown in the design guidelines:

image

**Comment by mmenestr** [comment]
Yes! When the card is only selectable, it has a blue border (seen here also https://staging.patternfly.org/components/card/design-guidelines#selectable-cards) but when the card is both selectable AND clickable, it does not (seen here: https://staging.patternfly.org/components/card/design-guidelines#selectable-and-clickable-cards) ! I think it was done this way to distinguish between the card being clicked vs selected. If the card is checked, we don't want it to have an outline because it'll feel like it's also been selected even though it might not have been.

**Comment by andrew-ronaldson** [comment]
Changes look good @jenny-s51, thanks!

File: packages/react-core/src/demos/PrimaryDetail.md
Comment (thatblindgeye) [reply packages/react-core/src/demos/PrimaryDetail.md:572]: FWIW, this is the current interaction behavior in the Primary Detail Card View demo:

https://github.com/patternfly/patternfly-react/assets/70952936/f06cfdfe-b972-4351-bef5-53138b2e0f1a

The Drawer opens unexpectedly when the kebab and checkbox are clicked (via mouse or keyboard). With those issues fixed, there'd then be no way to open the drawer via keyboard.

Comment (andrew-ronaldson) [reply packages/react-core/src/demos/PrimaryDetail.md:572]: This is a fun one! I see the demos for the full-page Primary details also have nested clickable content.
Short-term we could remove these interactive elements from the demo to align with the design guidance. 
Looking ahead the new cards have the title as the clickable element so we could let that trigger the drawer without interfering with the other clickable elements.

Comment (jenny-s51) [reply packages/react-core/src/demos/PrimaryDetail.md]: Thanks @andrew-ronaldson this is a great idea - updated the primary detail demo to set the title to a clickable link which triggers the drawer

File: packages/react-core/src/demos/PrimaryDetail.md
Comment (thatblindgeye) [inline packages/react-core/src/demos/PrimaryDetail.md:588]: The onCardClick may need to be tweaked a bit to get the `id` of the currentTarget's nearest Card (or the `id` from Card moved here if it's not being used for much else). The drawer content isn't getting rendered properly currently.

If we do this, we could also add an `aria-expanded` attribute on this button based on whether the `activeCard` state matches the `id`.

File: packages/react-core/src/demos/PrimaryDetail.md
Comment (thatblindgeye) [inline packages/react-core/src/demos/PrimaryDetail.md:545]: Should be able to remove this and where onKeyDown is declared now

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
The demos are looking great, thank you for the updates @jenny-s51!! Just had a few quick comments below

**Review by thatblindgeye — APPROVED** [review]
🎉 Thanks again for making these updates!

**Comment by mcoker** [comment]
@srambach would you mind taking a look at this one?

**Comment by mmenestr** [comment]
Just reviewed this and noticed the selected state of the selectable and clickable card still has a change in card outline. The intended design wasn't that - but maybe it's fine (?) @andrew-ronaldson what do you think? This was the design: 
image

and this is what's in the demo (see change in outline from selected to non selected)
image

**Review by jenny-s51 — COMMENTED** [review]
@mmenestr Yes looks like this divergence from the design appears in both the core and react workspaces currently, and is therefore out of scope of this original issue.

It seems like either we should create standalone issues to update core / react to match the existing design, or update the design guidelines to match the current behavior. 

@andrew-ronaldson which direction would you suggest we take with this?

**Comment by srambach** [comment]
I agree, the reimplementation of the clickable/selectable card put that border change on it. If design doesn't want it, it would be a core task to remove it.

**Comment by andrew-ronaldson** [comment]
@mmenestr @jenny-s51 @srambach I don't think the border change is needed for the **selected card** so we can make a Core issue to update the card to match the design guidelines.

**Comment by jenny-s51** [comment]
Thank you for your awesome feedback @srambach @andrew-ronaldson @mmenestr ! Opened an issue in core to update the card styling. 

When you have a moment, would be great if you could all kindly take a final look at this PR and approve if everything else looks good.

**Review by srambach — APPROVED** [review]
LGTM

**Review by mmenestr — APPROVED** [review]

**Review by andrew-ronaldson — APPROVED** [review]
LGTM, thanks!

**Review by tlabaj — APPROVED** [review]
lgtm

**Pull request merged**
